docs/okeps/okep-4456-egressip-on-user-defined-networks.md

```diff
@@ -0,0 +1,315 @@
+
+```
+sh-5.2# ip route show table 1111
+default dev eth1
+```
+
+The IPTable entry in OVN-KUBE-EGRESS-IP-MULTI-NIC chain would be changed to match only on packet mark and SNAT to the associated EgressIP address.
+This alleviates the need for IPTable entry per pod.
+
+```
+-A OVN-KUBE-EGRESS-IP-MULTI-NIC -m mark --mark 0xa -o eth1 -j SNAT --to-source 1.1.1.1
+```
+
+```
++----------------------------------------------------------------------------------------------------------------------------------------------------+
+|                                             secondaryeip-cdn-l3-ic and secondaryeip-udn-l3-ic                                                      |
++============================================+==================================================+===========================+========================+
+| Egress node is local,                      | Egress node is local,                            | Egress node is remote,    | Egress node is remote, |
+| pod is local                               | pod is remote                                    | pod is local              | pod is remote          |
++--------------------------------------------+--------------------------------------------------+---------------------------+------------------------+
+| Ovn_cluster_router                         | Ovn_cluster_router                               | Ovn_cluster_router        | Do nothing             |
+| - LRPs                                     | - LRPs                                           | - LRPs                    |                        |
+|   - Redirect to its mgnt port and pkt mark |   - Redirect to its mgnt port and pkt mark       |   - Redirect to remote ts |                        |
+| IP rules                                   | IP rules                                         | IP rules                  |                        |
+| - Match on mark and redirect to table      | - Match on mark and redirect to table            | IPTables                  |                        |
+| IPTables                                   | IPTables                                         |                           |                        |
+| - SNAT back on pkt mark                    | - SNAT back on pkt mark                          |                           |                        |
+| - Save mark for rpl                        | - Save mark for rpl                              |                           |                        |
++--------------------------------------------+--------------------------------------------------+---------------------------+------------------------+
+secondaryeip - EgressIP is node's secondary interface(s).
+```
+
+```
++-----------------------------------------------------------+
+| secondaryeip-cdn-l3-non-ic and secondaryeip-udn-l3-non-ic |
++===========================================================+
+| TBD                                                       |
```

Comment:
Options:
* Not support EIPs assigned to secondary interfaces for non-ic
* Use `ovn_cluster_router` to redirect to egress nodes GW and then also add a LRP to that egress nodes GW LRPs that redirect it to its mgnt port and pkt mark